### PR TITLE
Update text annotations to not use deprecated features

### LINKED ISSUE #4180: Remove deprecated Text.color attribute from graphical annotations
### Description
According to the Modelica specification section 18.6.5.5 a `text` graphical object only has the attribute `textColor` but not `color`.
However in the MSL this `color` attribute is used at the following location:

- `Modelica.Blocks.Discrete.TriggeredMax`  class saved in `Blocks\Discrete.mo` file on line 410 

### Expected Behavior
`color` is replaced by `textColor` for graphical objects of type `text`.

### Checked versions
MSL 4.0.0 (master branch) and maint/4.0.x branch.

Thanks for looking into this
Aaron Buntrock-Bosch Rexroth

### PR DESCRIPTION
**Pull request by HansOlsson**
Fixes #4180

Note that it both replaces textColor and removes deprecated attributes.
```
updateModelicaAnnotations("Modelica",renameTextColor=true)
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Clocked.ClockSignals.Clocks.Logical.PartialLogicalClock.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Clocked.ClockSignals.Interfaces.ClockInput.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Clocked.Examples.Systems.ControlledMixingUnit.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Clocked.Examples.Systems.Utilities.ComponentsThrottleControl.Engine.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Clocked.RealSignals.Periodic.FIRbyCoefficients.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Clocked.RealSignals.Periodic.MovingAverage.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.ComplexBlocks.ComplexMath.ComplexToReal.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Electrical.Analog.Examples.OpAmps.OpAmpCircuits.Buffer.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Electrical.Machines.Sensors.SinCosResolver.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Electrical.Polyphase.Examples.PolyphaseRectifier.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Magnetic.FundamentalWave.Examples.BasicMachines.InductionMachines.ComparisonPolyphase.IMC_DOL_CommonLeakage.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Magnetic.FundamentalWave.Examples.BasicMachines.InductionMachines.ComparisonPolyphase.IMC_DOL_Polyphase.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Magnetic.FundamentalWave.Examples.BasicMachines.InductionMachines.ComparisonPolyphase.IMS_Start_Polyphase.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Magnetic.FundamentalWave.Examples.BasicMachines.SynchronousMachines.ComparisonPolyphase.SMEE_Generator_Polyphase.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Magnetic.FundamentalWave.Examples.BasicMachines.SynchronousMachines.ComparisonPolyphase.SMPM_Inverter_Polyphase.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Magnetic.FundamentalWave.Examples.BasicMachines.SynchronousMachines.ComparisonPolyphase.SMR_Inverter_Polyphase.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Magnetic.QuasiStatic.FluxTubes.Basic.ElectroMagneticConverter.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Magnetic.QuasiStatic.FluxTubes.Examples.FixedShapes.CuboidSections.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Magnetic.QuasiStatic.FluxTubes.Interfaces.AbsoluteSensor.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Magnetic.QuasiStatic.FluxTubes.Interfaces.NegativeMagneticPort.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Magnetic.QuasiStatic.FluxTubes.Interfaces.PositiveMagneticPort.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Magnetic.QuasiStatic.FluxTubes.Interfaces.RelativeSensor.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Magnetic.QuasiStatic.FluxTubes.Sensors.Transient.Permeability.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Magnetic.QuasiStatic.FundamentalWave.Examples.BasicMachines.InductionMachines.IMC_Conveyor.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Magnetic.QuasiStatic.FundamentalWave.Examples.BasicMachines.InductionMachines.IMC_Initialize.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Magnetic.QuasiStatic.FundamentalWave.Examples.BasicMachines.InductionMachines.IMC_Transformer.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Magnetic.QuasiStatic.FundamentalWave.Examples.BasicMachines.InductionMachines.IMC_YD.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Magnetic.QuasiStatic.FundamentalWave.Examples.BasicMachines.InductionMachines.IMS_Characteristics.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Magnetic.QuasiStatic.FundamentalWave.Examples.BasicMachines.SynchronousMachines.SMPM_CurrentSource.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Magnetic.QuasiStatic.FundamentalWave.Examples.BasicMachines.SynchronousMachines.SMPM_Mains.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Magnetic.QuasiStatic.FundamentalWave.Examples.BasicMachines.SynchronousMachines.SMPM_OpenCircuit.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Magnetic.QuasiStatic.FundamentalWave.Examples.BasicMachines.SynchronousMachines.SMR_CurrentSource.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Mechanics.MultiBody.Visualizers.Rectangle.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Mechanics.Translational.Components.Mass.
 
Converting name Text.lineColor to Text.textColor
 in Modelica.Mechanics.Translational.Examples.Friction.
 
Converting name Text.lineColor to Text.textColor
 in Modelica.Mechanics.Translational.Examples.SignConvention.
 
Converting name Text.{fillColor,pattern,fillPattern} to nothing
 in Modelica.Units.SI.
```
(There might be few unneeded formatting changes, might clear up.)

### CODE REVIEW

File: Modelica/Magnetic/QuasiStatic/FluxTubes/package.mo [hunk 1]
```diff
@@ -11,8 +11,7 @@ package FluxTubes "Library for modelling of quasi-static electromagnetic devices
 This library is intended to provide models for the investigation of
 quasi-static electromagnetic devices with lumped magnetic networks.
 </p>
-</html>"),
-           Icon(coordinateSystem(preserveAspectRatio=false), graphics={
+</html>"), Icon(coordinateSystem(preserveAspectRatio=false), graphics={
```
Comment (HansOlsson) [inline Modelica/Magnetic/QuasiStatic/FluxTubes/package.mo:14]: I know that this change isn't needed. But the previous variant was plain weird.

File: Modelica/Magnetic/QuasiStatic/FundamentalWave/Examples/BasicMachines/SynchronousMachines/SMR_CurrentSource.mo [hunk 2]
```diff
@@ -205,8 +205,8 @@ model SMR_CurrentSource
         origin={0,60})));
   Modelica.Electrical.QuasiStatic.Polyphase.Sensors.VoltageQuasiRMSSensor voltageQuasiRMSSensorQS(m=m)
     annotation (Placement(transformation(extent={{-40,60},{-20,40}})));
-  Modelica.Electrical.QuasiStatic.Polyphase.Basic.Star starMQS(m=m) annotation
-    (Placement(transformation(
+  Modelica.Electrical.QuasiStatic.Polyphase.Basic.Star starMQS(m=m) annotation (
+     Placement(transformation(
```
Comment (HansOlsson) [inline Modelica/Magnetic/QuasiStatic/FundamentalWave/Examples/BasicMachines/SynchronousMachines/SMR_CurrentSource.mo:209]:
```suggestion
  Modelica.Electrical.QuasiStatic.Polyphase.Basic.Star starMQS(m=m) annotation
    (Placement(transformation(
```
Revert.

File: Modelica/Mechanics/MultiBody/Visualizers/Rectangle.mo [hunk 3]
```diff
@@ -38,7 +38,8 @@ protected
     final r_0=frame_a.r_0,
     final nu=nu,
     final nv=nv,
-    redeclare function surfaceCharacteristic = Advanced.SurfaceCharacteristics.rectangle (
+    redeclare function surfaceCharacteristic =
+        Advanced.SurfaceCharacteristics.rectangle (
```
Comment (HansOlsson) [inline Modelica/Mechanics/MultiBody/Visualizers/Rectangle.mo:42]:
```suggestion
    redeclare function surfaceCharacteristic = Advanced.SurfaceCharacteristics.rectangle (
```
Revert

File: Modelica/Mechanics/MultiBody/package.mo [hunk 4]
```diff
@@ -145,8 +145,8 @@ model World
     annotation (Dialog(tab="Defaults"));
 
   replaceable function gravityAcceleration =
-       Modelica.Mechanics.MultiBody.Forces.Internal.standardGravityAcceleration (
-           gravityType=gravityType, g=g*Modelica.Math.Vectors.normalizeWithAssert(n), mu=mu)
+       Modelica.Mechanics.MultiBody.Forces.Internal.standardGravityAcceleration
+        (  gravityType=gravityType, g=g*Modelica.Math.Vectors.normalizeWithAssert(n), mu=mu)
```
Comment (HansOlsson) [inline Modelica/Mechanics/MultiBody/package.mo:149]:
```suggestion
       Modelica.Mechanics.MultiBody.Forces.Internal.standardGravityAcceleration (
           gravityType=gravityType, g=g*Modelica.Math.Vectors.normalizeWithAssert(n), mu=mu)
```
Revert

**Comment by beutlich** [comment]
> Converting name Text.lineColor to Text.textColor
>  in Modelica.Mechanics.Translational.Examples.Friction.
>  
> Converting name Text.lineColor to Text.textColor
>  in Modelica.Mechanics.Translational.Examples.SignConvention.

Thanks for fixing. I adapted the CI check by #4183 to also detect such corner cases.

File: Modelica/Magnetic/QuasiStatic/FluxTubes/package.mo (hunk 1 above)
Comment (tobolar) [inline Modelica/Magnetic/QuasiStatic/FluxTubes/package.mo:14]:
```suggestion
</html>"),
    Icon(coordinateSystem(preserveAspectRatio=false), graphics={
```

Comment (tobolar) [reply Modelica/Magnetic/QuasiStatic/FluxTubes/package.mo:14]: You're right, but there is plenty of such cases in the MSL code. :-(

**Review by beutlich — APPROVED** [review]
LGTM.

**Review by tobolar — APPROVED** [review]
Fine to me

File: Modelica/Magnetic/QuasiStatic/FluxTubes/package.mo (hunk 1 above)
Comment (HansOlsson) [reply Modelica/Magnetic/QuasiStatic/FluxTubes/package.mo:14]: I think it is good, but would prefer a general cleanup of those.

**Comment by beutlich** [comment]
Please note, that I pushed new changes to finally fix #4180 as reported by @abuntrock.

**Comment by HansOlsson** [comment]
> Please note, that I pushed new changes to finally fix #4180 as reported by @abuntrock.

I was wondering why the conversion didn't catch this, until I noticed that it wasn't named "lineColor", but "color" which has never been correct.

File: Modelica/Magnetic/QuasiStatic/FluxTubes/package.mo (hunk 1 above)
Comment (HansOlsson) [reply Modelica/Magnetic/QuasiStatic/FluxTubes/package.mo:14]: Ah well, adding it anyway.

**Comment by tobolar** [comment]
> (I guess @tobolar being a co-author on the last one is the issue, but I'm not sure.)

@HansOlsson Seems you are right. I supposed it was due to @beutlich but "re-requesting" my own review resolved the lock. I guess you can merge this now.

**Pull request merged**